search/search_repository_blackbox_test.go

```diff
@@ -58,102 +61,213 @@ func (s *searchRepositoryBlackboxTest) getTestFixture() *tf.TestFixture {
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, res[0].Fields["system.order"], wi2.Fields["system.order"])
+			assert.Equal(t, res[1].Fields["system.order"], wi1.Fields["system.order"])
+		})
+	})
+
+	s.T().Run("Filter with limits", func(t *testing.T) {
+
+		t.Run("none", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, nil, nil, nil)
+			// when
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 2, len(res))
+		})
+
+		t.Run("Filter with offset", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			start := 3
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, nil, &start, nil)
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 0, len(res))
+		})
+
+		t.Run("Filter with limit", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			limit := 1
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, nil, nil, &limit)
+			// then
+			require.Nil(s.T(), err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 1, len(res))
+		})
+	})
+
+	s.T().Run("Filter with parent-exists filter", func(t *testing.T) {
+
+		t.Run("no link created", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			parentExists := false
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, &parentExists, nil, nil)
+			// then both work items should be returned
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 2, len(res))
+		})
+
+		t.Run("link created", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture(tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinks[0].LinkTypeID = fxt.WorkItemLinkTypes[0].ID
+				fxt.WorkItemLinks[0].SourceID = fxt.WorkItems[0].ID
+				fxt.WorkItemLinks[0].TargetID = fxt.WorkItems[1].ID
+				return nil
+			}))
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			parentExists := false
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, &parentExists, nil, nil)
+			// then only parent work item should be returned
+			require.Nil(t, err)
+			assert.Equal(t, uint64(1), count)
+			assert.Equal(t, 1, len(res))
+			assert.Equal(t, fxt.WorkItems[0].ID, res[0].ID)
+		})
+
+		t.Run("link deleted", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture(tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
```

Comment:
Correct me if I'm wrong but here you should only need `fxt := tf.NewTestFixture(s.DB(), t, tf.WorkItemLinks(1))`. Then you can delete that link and test that it will not be returned. No need for the gigantic test fixture ;)

Reply:
not exactly:  the link type needs to be a `parent-of` in this test. But I simplified it a bit, nonetheless